latex/Inhalt/grundlagen.tex

```diff
@@ -0,0 +1,193 @@
+
+Calls hingegen entsprechen grob dem Funktionsaufruf aus higher-level Sprachen. Beim Aufruf einer Call-Instruktion wird die momentane Position im Code (der PC) auf dem Stack abgelegt, dies ist die sogenannte Return-Addresse, und anschließend ein Sprung zur angegebenen Addresse (zum Funktions-Code) ausgeführt. Die Return-Instruktion stellt das Ende einer Funktion dar, sie setzt den PC auf das oberste Element des Stacks, bei korrekter Verwendung ist dies die Return-Addresse.
+
+\begin{table}[h]
+    \centering
+    \caption{Beispielhafte Befehle für Verzweigungen}
+    \label{tab:jmp}
+    \begin{tabular}{l | l}
+        \asm{JMP 0FFH} & Setze PC auf 0FFH\\
+        \asm{JZ  0FFH} & Setze PC auf 0FFH falls Zero-Flag gesetzt ist \\
+        \asm{CALL 0FFH} & Speichere PC auf dem Stack und setze ihn auf FFH \\
+        \asm{RET} & Setze PC auf oberstes Stack Element (und entferne es) \\
+    \end{tabular}
+\end{table}
+
+\subsubsection{Arithmetik}
+
+Der Intel 8080 implementiert eine 8-Bit Arithmetik. Dafür werden die Bytes als Zahlen im Zweierkomplement interpretiert. Die Ergebnisse der Operationen werden hauptsächlich im Akkumulator gespeichert --- Ausnahmen sind \zB Inkrementierungen der Register. Viele der Operationen bearbeiten auch die Flaggen, um Informationen über das Ergebnis zurückzugeben (siehe \cref{sec:flags} für Details).
+\Cref{tab:arith} zeigt einige arithmetische Anweisungen mit einer kurzen Erklärung.
+
+\begin{table}[h]
+    \centering
+    \caption{Beispielhafte Befehle für arithmetische Operationen}
+    \label{tab:arith}
+    \begin{tabular}{l | l}
+        \asm{ADD B} & Addiere B auf Akkumulator\\
+        \asm{CMP B} & Setze Zero-Flag falls $A = B$, sonst setze Zero-Flag zurück\\
+        \asm{INR B} & Inkrementiere B um 1 \\
+    \end{tabular}
+\end{table}
+
+\subsection{Interrupts}
+
+Durch sogenannte Interrupts kann die normale Ausführung der CPU unterbrochen werden. Interrupts sind externe Signale (bspw. von Peripheriegeräten), bei denen es sich um eine einzelne Instruktion handelt. Diese Instruktion wird ausgeführt, anschließend wird die normale Ausführung fortgesetzt. Meistens handelt es sich bei dieser Instruktion um eine \asm{RST}-Instruktion, eine Gruppe von Instruktionen, die \asm{CALL}s auf fixe Addressen realisieren. Dadurch kann der Entwickler den entsprechenden Code um die Interrupts zu bearbeiten an diesen Addressen ablegen, sodass er somit ausgeführt werden kann.
+
+Der Entwickler kann entscheiden ob Interrupts möglich sind, mithilfe der \asm{EI} (Enable Interrupts) bzw. \asm{DI} (Disable Interrupts) Anweisungen. Diese setzen eine interne Flagge, die bestimmt ob Interrupts zugelassen sind. Interrupts während diese Flagge \rust{false} ist werden ignoriert. Bei Programmstart ist dies der Fall.
```

Comment:
I'm not sure if there's a definition on how to do it correctly, but I think it should be "DI-Anweisung" as well as "EI-" or "mithilfe der Anweisungen EI [...]"

Also we really shouldn't translate "Flag" if we keep "Interrupt"

"während der Wert des Flags false ist"